modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/GlassViewEventHandler.java

```diff
@@ -669,9 +669,12 @@ protected void onChanged(ListChangeListener.Change<InputMethodTextRun> c) {
     @Override
     public double[] getInputMethodCandidatePos(int offset) {
         Point2D p2d = scene.inputMethodRequests.getTextLocation(offset);
+        final Screen s = Screen.getMainScreen();
+        float pScaleX = (s == null) ? 1.0f : s.getPlatformScaleX();
+        float pScaleY = (s == null) ? 1.0f : s.getPlatformScaleY();
         double[] ret = new double[2];
-        ret[0] = p2d.getX();
-        ret[1] = p2d.getY();
+        ret[0] = p2d.getX() * pScaleX;
+        ret[1] = p2d.getY() * pScaleY;
```

Comment:
You should use `Screen::toPlatformX/Y` or it won't work for all screen arrangements.